### DISCUSSION TITLE
theme ui with Typescript

### DISCUSSION THREAD

**Discussion by igotfr** [system-ui/theme-ui#1161] [General] [answered]
Type '{ children: Element[]; sx: { backgroundColor: string; borderRadius: number; fontSize: number; margin: number; padding: number; }; }' is not assignable to type 'DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>'.

  Property 'sx' does not exist on type 'DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>'.ts(2322)

how to solve this?

**Comment by jorisw** [comment]
ThemeUI expects you to exclusively use ThemeUI components, at least when you're passing the `sx` prop.

So replace your `<div />` with a `Flex` or `Box` component.

**Reply by igotfr** [reply]
the problem occurs exclusively with .tsx, on .js files works normally

**Reply by jorisw** [reply]
That's because... only `ts` and `tsx` files are treated as TypeScript and therefore typechecked.

**Reply by igotfr** [reply]
is not there a way to use common html tags with sx and Typescript?

**Reply by hasparus** [reply]
> ThemeUI expects you to exclusively use ThemeUI components, at least when you're passing the sx prop.

We're adding `sx` to global JSX types, so you can use it with JSX pragma with any "HTML tag".

Take a look here: https://github.com/system-ui/theme-ui/blob/master/packages/core/src/react-jsx.ts

@cindRoberta which version of Theme UI are you using?

**Reply by igotfr** [reply]
0.3.1

**Comment by hasparus** [comment, marked as answer]
It should _Just Work(TM)_. Here's a CodeSandbox.

https://codesandbox.io/s/vibrant-goldwasser-cfenj?file=/src/App.tsx

I'll try to help you out, but I think we need more details.

**Reply by igotfr** [reply]
solution: npm i -D @types/theme-ui
